integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -296,7 +339,11 @@ def from_dict(cls, data: Dict[str, Any]) -> "AzureAISearchDocumentStore":
             Deserialized component.
         """
         if (fields := data["init_parameters"]["metadata_fields"]) is not None:
-            data["init_parameters"]["metadata_fields"] = cls._deserialize_metadata_fields(fields)
```

Comment:
It looks like `_deserialize_metadata_fields` is not used anymore right? If so let's go ahead and remove the function